### PR TITLE
fixed using wrong package manager in yarn workspace projects

### PR DESCRIPTION
**Pull request by connectdotz**
this is to fix react-native-scripts not able to find `yarn.lock` file in the monorepo (workspaces) projects, which caused the react-native-scripts to use the wrong package manager and failed.  

yarn workspace project puts the` yarn.lock` in the project root instead of the react native app directory. Since the current implementation only looks for yarn.lock in the app/cwd directory, it doesn't see the file and will wrongly select npm as the package manager. The new lookup method will traverse up the filesystem (similar to node module resolution) until it either finds the file or hits the filesystem root. 

**outline the changes:**

1. consolidated all yarn.lock lookup into a new file  `react-native-scripts/src/utils/pm.js`, which performs the new logic described above. 
1. setup jest testing. Future contributors can run `yarn test` to guard against regression. Also updated `taskfile.js` to exclude tests in the build.
1. added test for the new logic

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,9 +1,20 @@
...
+    "format": "prettier --single-quote --trailing-comma=es5 --print-width=100 --write '{create-react-native-app,react-native-scripts}/{src,template}/**/*.js'",
+    "test": "jest"
   },
   "devDependencies": {
+    "jest": "^21.2.1",
     "prettier": "^0.21.0"
+  },
+  "jest": {
+    "moduleFileExtensions": [ "js" ],
+    "testRegex": "(/__tests__/.*|\\.(test|spec))\\.(js)$",
```
Comment (fson) [inline package.json]: I believe these are the default settings in Jest now, so we should be able to omit `moduleFileExtensions` and `testRegex`.

File: package.json [hunk 2]
```diff
@@ -1,9 +1,20 @@
...
+    "testPathIgnorePatterns": [
+      "\\.snap$",
```
Comment (fson) [inline package.json]: I think this can be omitted.

**Comment by connectdotz** [comment]
done

**Comment by connectdotz** [comment]
ping

**Comment by fson** [comment]
Hey, looks like something went wrong with merging changes from master. Could you please rebase this branch onto master?

**Comment by connectdotz** [comment]
rebased

**Comment by fson** [comment]
Thanks @connectdotz!

**Pull request merged**